storage/kvstore/source/TDBStore.cpp

```diff
@@ -1041,7 +1042,7 @@ int TDBStore::init()
     }
 
     _prog_size = _bd->get_program_size();
-    _work_buf = new uint8_t[work_buf_size];
+    _work_buf = new uint8_t[_prog_size];
```

Comment:
As above, we read `record_header_t` (a 32-bit aligned struct of size 24 bytes) into `_work_buff`, but
* here it's only 8-bit aligned
* the program size can be less than that - some flashes can be programmed to 1-byte, in this case the buffer is insufficient

This could be why the Greentea test failed and hard faulted.